cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go

```diff
@@ -114,7 +114,16 @@ func NewCStorVolumeReplicaController(
 			q.Operation = common.QOpAdd
 			glog.Infof("cStorVolumeReplica Added event : %v, %v", cVR.ObjectMeta.Name, string(cVR.ObjectMeta.UID))
 			controller.recorder.Event(cVR, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageCreateSynced))
-			cVR.Status.Phase = apis.CVRStatusPending
+
+			// For New request phase of cVR will be empty
+			// ToDO: Need to have an annotation in CSP and CVR which will state
+			// about recreation events.
+			if IsEmptyStatus(cVR) {
+				cVR.Status.Phase = apis.CVRStatusInit
```

Comment:
One improvement here is: If volume creation fails in the first time, and the status got set to 'Error'.. and, before the volume gets successfully created if the node gets replaced, it will be marked as 'Recreate' which won't cause any harm, but, it can be corrected (or) can be left out as a comment here.

Reply:
No @vishnuitta before lines in the code are saying if the phase is set to `Error` We are not performing any events.